fix: (storagetransfer) transfer path can be changed to empty string

gurusai-voleti opened the pull request:
<!--
Complete the self-review checklist to help speed up the review process: https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/


If your PR is still work in progress, please create it in draft mode.

Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to.
For example: Fixes https://github.com/hashicorp/terraform-provider-google/issues/ISSUE_ID
-->
**Release Note Template for Downstream PRs (will be copied)**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/13673
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/20421
See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:bug
storagetransfer: fix to allow empty string for path
```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12368-old..auto-pr-12368) ( 1 file changed, 12 insertions(+), 2 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12368-old..auto-pr-12368) ( 1 file changed, 12 insertions(+), 2 deletions(-))

## Breaking Change(s) Detected

The following breaking change(s) were detected within your pull request.
- Field `transfer_spec.azure_blob_storage_data_source.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)
- Field `transfer_spec.gcs_data_sink.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)
- Field `transfer_spec.gcs_data_source.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)

If you believe this detection to be incorrect please raise the concern with your reviewer.
If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
An `override-breaking-change` label can be added to allow merging.



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_storage_transfer_job` (25 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_storage_transfer_job" "primary" {
  transfer_spec {
    azure_blob_storage_data_source {
      path = # value needed
    }
  }
}


```

modular-magician commented:
#### Tests analytics
Total tests: 11
Passed tests: 11
Skipped tests: 0
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>storagetransfer</li>

</ul>

</blockquote>
</details>

🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12368/artifacts/44c4d047-6b9d-4ae4-bc8c-14ec01066c03/build-log/replaying_test.log)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12368-old..auto-pr-12368) ( 1 file changed, 6 insertions(+), 1 deletion(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12368-old..auto-pr-12368) ( 1 file changed, 6 insertions(+), 1 deletion(-))

## Breaking Change(s) Detected

The following breaking change(s) were detected within your pull request.
- Field `transfer_spec.gcs_data_sink.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)
- Field `transfer_spec.gcs_data_source.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)

If you believe this detection to be incorrect please raise the concern with your reviewer.
If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
An `override-breaking-change` label can be added to allow merging.

modular-magician commented:
#### Tests analytics
Total tests: 11
Passed tests: 11
Skipped tests: 0
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>storagetransfer</li>

</ul>

</blockquote>
</details>

🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12368/artifacts/14affe89-342f-4132-83a8-d731c6cfcae6/build-log/replaying_test.log)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 122 insertions(+), 4 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 122 insertions(+), 4 deletions(-))

## Breaking Change(s) Detected

The following breaking change(s) were detected within your pull request.
- Field `transfer_spec.gcs_data_sink.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)
- Field `transfer_spec.gcs_data_source.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)

If you believe this detection to be incorrect please raise the concern with your reviewer.
If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
An `override-breaking-change` label can be added to allow merging.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 122 insertions(+), 4 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 122 insertions(+), 4 deletions(-))

## Breaking Change(s) Detected

The following breaking change(s) were detected within your pull request.
- Field `transfer_spec.gcs_data_sink.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)
- Field `transfer_spec.gcs_data_source.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)

If you believe this detection to be incorrect please raise the concern with your reviewer.
If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
An `override-breaking-change` label can be added to allow merging.

modular-magician commented:
#### Tests analytics
Total tests: 12
Passed tests: 11
Skipped tests: 0
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>storagetransfer</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccStorageTransferJob_transferPathError</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccStorageTransferJob_transferPathError` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12368/artifacts/c6f1f74c-8554-4f69-8f07-d85f66120e36/recording/TestAccStorageTransferJob_transferPathError.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---






🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12368/artifacts/c6f1f74c-8554-4f69-8f07-d85f66120e36/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-12368/artifacts/c6f1f74c-8554-4f69-8f07-d85f66120e36/recording) for each test

modular-magician commented:
#### Tests analytics
Total tests: 12
Passed tests: 11
Skipped tests: 0
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>storagetransfer</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccStorageTransferJob_transferPathError</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

kautikdk commented:
Please link the issue in PR description that you are resolving.

gurusai-voleti commented:
> Hi there, I'm the Modular magician. I've detected the following information about your changes:
> 
> ## Diff report
> Your PR generated some diffs in downstreams - here they are.
> 
> `google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 122 insertions(+), 4 deletions(-)) `google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 122 insertions(+), 4 deletions(-))
> 
> ## Breaking Change(s) Detected
> The following breaking change(s) were detected within your pull request.
> 
> * Field `transfer_spec.gcs_data_sink.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)
> * Field `transfer_spec.gcs_data_source.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)
> 
> If you believe this detection to be incorrect please raise the concern with your reviewer. If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window. An `override-breaking-change` label can be added to allow merging.

Path is set as computed field which is user provided i see it setup incorrectly as a computed field, so removed it now and added validation as GCS bucket paths cannot be started with / and added diffsuppressfunc to revert to "" empty string as well

kautikdk commented:
Can you please briefly explain your solution and how it will allow users to set empty string in path field?

gurusai-voleti commented:
> Can you please briefly explain your solution and how it will allow users to set empty string in path field?

Path is an optional field, if not provided at first tf apply it is set to empty string,
but in the customer reported issue they used "/" first and later realized to change to "" empty string
which they cannot as in update d.HasChanges will not detect empty string as a change

so added DiffSuppressFunc if the old and new is same it will not show any difference in tf plan, if its different it will show in tf plan and path can be updated to "" empty string and added validation to the path field as gcs object path cannot be started with "/" (preference here https://cloud.google.com/storage-transfer/docs/reference/rest/v1/GcsData)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 127 insertions(+), 4 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 127 insertions(+), 4 deletions(-))

## Breaking Change(s) Detected

The following breaking change(s) were detected within your pull request.
- Field `transfer_spec.gcs_data_sink.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)
- Field `transfer_spec.gcs_data_source.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)

If you believe this detection to be incorrect please raise the concern with your reviewer.
If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
An `override-breaking-change` label can be added to allow merging.

modular-magician commented:
#### Tests analytics
Total tests: 12
Passed tests: 12
Skipped tests: 0
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>storagetransfer</li>

</ul>

</blockquote>
</details>

🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12368/artifacts/72e36c2f-a49f-48eb-80d3-0492f6316dd4/build-log/replaying_test.log)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 129 insertions(+), 4 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 129 insertions(+), 4 deletions(-))

## Breaking Change(s) Detected

The following breaking change(s) were detected within your pull request.
- Field `transfer_spec.gcs_data_sink.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)
- Field `transfer_spec.gcs_data_source.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)

If you believe this detection to be incorrect please raise the concern with your reviewer.
If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
An `override-breaking-change` label can be added to allow merging.

modular-magician commented:
#### Tests analytics
Total tests: 12
Passed tests: 12
Skipped tests: 0
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>storagetransfer</li>

</ul>

</blockquote>
</details>

🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12368/artifacts/cec81f08-a8d4-4104-b31c-f6b587d42046/build-log/replaying_test.log)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 124 insertions(+), 4 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 124 insertions(+), 4 deletions(-))

## Breaking Change(s) Detected

The following breaking change(s) were detected within your pull request.
- Field `transfer_spec.gcs_data_sink.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)
- Field `transfer_spec.gcs_data_source.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)

If you believe this detection to be incorrect please raise the concern with your reviewer.
If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
An `override-breaking-change` label can be added to allow merging.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 225 insertions(+), 4 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 225 insertions(+), 4 deletions(-))

## Breaking Change(s) Detected

The following breaking change(s) were detected within your pull request.
- Field `transfer_spec.gcs_data_sink.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)
- Field `transfer_spec.gcs_data_source.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)

If you believe this detection to be incorrect please raise the concern with your reviewer.
If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
An `override-breaking-change` label can be added to allow merging.

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccStorageTransferJob_transferUpdateToEmptyString` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12368/artifacts/411c3aaa-2738-4e52-9503-99a043c9ff7c/recording/TestAccStorageTransferJob_transferUpdateToEmptyString.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---






🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12368/artifacts/411c3aaa-2738-4e52-9503-99a043c9ff7c/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-12368/artifacts/411c3aaa-2738-4e52-9503-99a043c9ff7c/recording) for each test

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 225 insertions(+), 4 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12368-old..auto-pr-12368) ( 2 files changed, 225 insertions(+), 4 deletions(-))

## Breaking Change(s) Detected

The following breaking change(s) were detected within your pull request.
- Field `transfer_spec.gcs_data_sink.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)
- Field `transfer_spec.gcs_data_source.path` transitioned from optional+computed to optional `google_storage_transfer_job` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes#field-oc-to-c)

If you believe this detection to be incorrect please raise the concern with your reviewer.
If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
An `override-breaking-change` label can be added to allow merging.

modular-magician commented:
#### Tests analytics
Total tests: 13
Passed tests: 13
Skipped tests: 0
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>storagetransfer</li>

</ul>

</blockquote>
</details>

🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12368/artifacts/ff6240dc-ca8c-4a15-bede-9927cb15372c/build-log/replaying_test.log)

modular-magician commented:
#### Tests analytics
Total tests: 13
Passed tests: 13
Skipped tests: 0
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>storagetransfer</li>

</ul>

</blockquote>
</details>

🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12368/artifacts/7fd3ae03-dbb3-4eb8-bd39-3a3d7aac5715/build-log/replaying_test.log)

kautikdk commented:
This appears to be breaking change as we removing computed from optional+computed so I think this should be scheduled for upcoming major release?

gurusai-voleti commented:
> This appears to be breaking change as we removing computed from optional+computed so I think this should be scheduled for upcoming major release?

I think it can be scheduled to be in near major release

modular-magician commented:
#### Tests analytics
Total tests: 13
Passed tests: 13
Skipped tests: 0
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>storagetransfer</li>

</ul>

</blockquote>
</details>

🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12368/artifacts/76549191-e6ea-4688-895a-71c34fda359a/build-log/replaying_test.log)

kautikdk commented:
If this is planned for major release then I think we should mark this PR draft or we can close it for now until guidelines for next major release is available.

kautikdk commented:
Hi @gurusai-voleti, Can you please make this PR draft and update internal ticket for the same? because we are not actively looking into this change until next major release which is not going to happen soon.

gurusai-voleti commented:
> Hi @gurusai-voleti, Can you please make this PR draft and update internal ticket for the same? because we are not actively looking into this change until next major release which is not going to happen soon.

updated on task and converted to draft PR

modular-magician commented:
#### Tests analytics
Total tests: 17
Passed tests: 17
Skipped tests: 0
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>storagetransfer</li>

</ul>

</blockquote>
</details>
🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12368/artifacts/7a2821af-d90c-4cf2-8f22-de0e014e7bb8/build-log/replaying_test.log)

gurusai-voleti commented:
closing as this has been done 14377
